snuba/consumers/consumer_builder.py

```diff
@@ -75,31 +122,38 @@ def __init__(
         validate_schema: bool = False,
         profile_path: Optional[str] = None,
     ) -> None:
-        self.storage = get_writable_storage(storage_key)
+        self.storages = {key: get_writable_storage(key) for key in storage_keys}
         self.bootstrap_servers = kafka_params.bootstrap_servers
         self.consumer_group = kafka_params.group_id
-        topic = (
-            self.storage.get_table_writer()
-            .get_stream_loader()
-            .get_default_topic_spec()
-            .topic
-        )
+
+        # We establish that there is only one input raw topic,
+        # one commit log topic, and one replacements topic amongst the storages
+        # We retain the raw logical topic for producer/consumer configuration
+        self.__topic = verify_single_topic([*self.storages.values()])
 
         # Ensure that the slice, storage set combination is valid
-        validate_passed_slice(self.storage.get_storage_set_key(), slice_id)
+        for writable_storage in self.storages.values():
+            validate_passed_slice(writable_storage.get_storage_set_key(), slice_id)
 
         self.broker_config = get_default_kafka_configuration(
-            topic, slice_id, bootstrap_servers=kafka_params.bootstrap_servers
+            SnubaTopic(self.__topic),
+            slice_id,
+            bootstrap_servers=kafka_params.bootstrap_servers,
         )
         logger.info(f"librdkafka log level: {self.broker_config.get('log_level', 6)}")
 
-        stream_loader = self.storage.get_table_writer().get_stream_loader()
+        # Get a random storage from the storage dict
+        rand_storage = [*self.storages.values()][0]
```

Comment:
It doesn't seem very random... isn't it just the first one?

Reply:
Oops, I forgot to update the comment and names from earlier